docs/reference/query-dsl/rule-query.asciidoc

```diff
@@ -0,0 +1,85 @@
+[role="xpack"]
+[[query-dsl-rule-query]]
+=== Rule Query
+
+preview::[]
+
+Applies rules configured in the <<query-rules-apis, Query Rules API>> to the query.
+This feature is used to promote documents in the manner of a <<query-dsl-pinned-query>> based on matching defined rules.
+If no matching query rules are defined, the "organic" matches for the query are returned.
+
+==== Example request
+
+////
+
+[source,console]
+--------------------------------------------------
+PUT _query_rules/my-ruleset
+{
+    "rules": [
+        {
+            "rule_id": "my-rule1",
+            "type": "pinned",
+            "criteria": [
+                {
+                    "type": "exact",
+                    "metadata": "query_string",
+                    "value": "pugs"
+                }
+            ],
+            "actions": {
+                "ids": [
+                    "id1"
+                ]
+            }
+        }
+    ]
+}
+--------------------------------------------------
+// TESTSETUP
+
+[source,console]
+--------------------------------------------------
+DELETE _query_rules/my-ruleset
+--------------------------------------------------
+// TEARDOWN
+
+////
+
+[source,console]
+--------------------------------------------------
+GET /_search
+{
+  "query": {
+    "rule_query": {
+      "match_criteria": {
+        "query_string": "pugs"
+      },
+      "ruleset_id": "my-ruleset",
+      "organic": {
+        "match": {
+          "description": "pugs"
+        }
+      }
+    }
+  }
+}
+--------------------------------------------------
+
+[[rule-query-top-level-parameters]]
+==== Top-level parameters for `rule_query`
+
+`ruleset_id`::
+(Required, string) <<mapping-id-field, Document IDs>> <<query-rules-apis, query ruleset>> ID with query-based rules to match and apply as applicable.
+`match_criteria`::
+(Required, object) Defines the match criteria to apply to rules in the given query ruleset.
+
+The following `match_criteria` are supported:
```

Comment:
This section renders a bit weird, `query_string` doesn't seem to be nested under `match_criteria`:

![image](https://github.com/elastic/elasticsearch/assets/6339205/8a766abc-9e3e-46fc-9540-93f7868e410b)

Reply:
Good catch, thank you! Fixed (screenshot added to PR description)